src/ui/common/src/components/pages/workflow/id/hook.ts

```diff
@@ -0,0 +1,109 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
+
+import { BreadcrumbLink } from '../../../../components/layouts/NavBar';
+import { handleGetWorkflowDag } from '../../../../handlers/getWorkflowDag';
+import { handleGetWorkflowDagResult } from '../../../../handlers/getWorkflowDagResult';
+import { WorkflowDagResultWithLoadingStatus } from '../../../../reducers/workflowDagResults';
+import { WorkflowDagWithLoadingStatus } from '../../../../reducers/workflowDags';
+import { AppDispatch, RootState } from '../../../../stores/store';
+import { getPathPrefix } from '../../../../utils/getPathPrefix';
+import { isInitial } from '../../../../utils/shared';
+
+export type useWorkflowOutputs = {
+  breadcrumbs: BreadcrumbLink[];
+  workflowId: string;
+  workflowDagId: string;
+  workflowDagResultId: string;
+  workflowDagWithLoadingStatus: WorkflowDagWithLoadingStatus;
+  workflowDagResultWithLoadingStatus: WorkflowDagResultWithLoadingStatus;
+};
+
+export default function useWorkflow(
```

Comment:
handles retrievals of `wfDag` and `wfDagResults`, including all `useEffect`s that triggers API calls when those two objects are not already fetched.